### PR TITLE
Add faster Price Tools

### PR DESCRIPTION
**Pull request by vovac12**

### CODE REVIEW

File: pallets/price-tools/src/migration.rs [hunk 1]
```diff
@@ -1,20 +1,22 @@
-use frame_support::dispatch::GetStorageVersion;
-use frame_support::traits::StorageVersion;
+pub mod v2 {
```
[the remark this answers is not in the document]
Comment (vovac12) [reply pallets/price-tools/src/migration.rs]: Removed

File: pallets/price-tools/src/migrations/v2.rs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        PriceInfos::<T>::translate::<PriceInfo, _>(
+            |_, old_price_info| -> Option<AggregatedPriceInfo> {
+                Some(AggregatedPriceInfo {
+                    buy: old_price_info,
+                    sell: PriceInfo::default(),
+                })
+            },
+        );
+        StorageVersion::new(2).put::<Pallet<T>>()
+    }
```
Comment (wer1st) [inline pallets/price-tools/src/migrations/v2.rs:49]: Storage version isn't changed: 2 -> 2
Is it ok?

Comment (vovac12) [reply pallets/price-tools/src/migrations/v2.rs:49]: It checks that version is lower than 2 and then set it to 2, so it's ok

File: pallets/qa-tools/src/tests/mcbc.rs [hunk 3]
```diff
@@ -31,6 +31,7 @@
 use super::{alice, register_custom_asset, QaToolsPallet};
 use assets::AssetIdOf;
 use common::prelude::{BalanceUnit, QuoteAmount};
+use common::PriceToolsProvider;
 use common::{
     assert_approx_eq, balance, AccountIdOf, AssetInfoProvider, Balance, DEXId, LiquiditySource,
     PriceVariant, CERES_ASSET_ID, ETH, TBCD, VAL, XOR,
```
[the remark this answers is not in the document]
Comment (wer1st) [reply pallets/qa-tools/src/tests/mcbc.rs]: And what about "below" situations?

File: pallets/qa-tools/src/tests/price_tools.rs [hunk 4]
```diff
@@ -30,6 +30,7 @@
 
 use super::QaToolsPallet;
 use assets::AssetIdOf;
+use common::PriceToolsProvider;
 use common::{balance, PriceVariant, ETH, XOR};
```
Comment (wer1st) [inline pallets/qa-tools/src/tests/price_tools.rs:34]:
```suggestion
use common::{balance, PriceToolsProvider, PriceVariant, ETH, XOR};
```

File: pallets/qa-tools/src/tests/xst.rs [hunk 5]
```diff
@@ -30,6 +30,7 @@
 
 use super::{alice, QaToolsPallet};
 use common::prelude::{err_pays_no, BalanceUnit, QuoteAmount, SwapVariant};
+use common::PriceToolsProvider;
 use common::{
     assert_approx_eq, balance, fixed, AssetId32, AssetIdOf, AssetName, AssetSymbol, DEXId,
     LiquiditySource, PredefinedAssetId, PriceVariant, SymbolName, XOR,
```
Comment (wer1st) [inline pallets/qa-tools/src/tests/xst.rs:36]:
```suggestion
use common::{
    assert_approx_eq, balance, fixed, AssetId32, AssetIdOf, AssetName, AssetSymbol, DEXId,
    LiquiditySource, PredefinedAssetId, PriceToolsProvider, PriceVariant, SymbolName, XOR,
```

File: pallets/xst/src/tests.rs [hunk 6]
```diff
@@ -49,6 +49,7 @@ mod tests {
     use sp_arithmetic::traits::Zero;
     use sp_arithmetic::FixedU128;
     use sp_std::collections::vec_deque::VecDeque;
+    use common::PriceToolsProvider;
```
Comment (wer1st) [inline pallets/xst/src/tests.rs:52]: Move it to line 40

File: runtime/src/xor_fee_impls.rs [hunk 7]
```diff
@@ -34,6 +34,7 @@ use bridge_types::{traits::EVMBridgeWithdrawFee, GenericNetworkId};
 #[cfg(feature = "wip")] // Dynamic fee
 use common::prelude::FixedWrapper;
 use common::LiquidityProxyTrait;
+use common::PriceToolsProvider;
```
Comment (wer1st) [inline runtime/src/xor_fee_impls.rs:37]:
```suggestion
use common::{LiquidityProxyTrait, PriceToolsProvider};
```

**Review by wer1st — APPROVED** [review]

**Review by Alexey-N-Chernyshov — APPROVED** [review]

**Pull request merged**